src/utils/getUniqueIdGenerator.js

```diff
@@ -0,0 +1,8 @@
+function getUniqueIdGenerator() {
+  let index = 0;
+  return () => {
+    return `mui-id-${index++}`;
```

Comment:
I don't think that it would work. Have a look at https://github.com/callemall/material-ui/issues/3533.